src/modules/0.0.22/aws_elasticache/index.ts

```diff
@@ -112,36 +109,38 @@ class CacheClusterMapper extends MapperBase<CacheCluster> {
       }
       return out;
     },
-    read: async (ctx: Context, clusterId?: string) => {
+    read: async (ctx: Context, id?: string) => {
       const enabledRegions = (await ctx.getEnabledAwsRegions()) as string[];
-      if (clusterId) {
-        for (const region of enabledRegions) {
+      if (!!id) {
+        const [clusterId, region] = id.split('|');
+        if (enabledRegions.includes(region)) {
           const client = (await ctx.getAwsClient(region)) as AWS;
           const rawCluster = await this.getCacheCluster(client.elasticacheClient, clusterId);
-          if (!rawCluster) continue;
-          if (rawCluster?.CacheClusterStatus === 'deleting') continue;
-
-          return this.cacheClusterMapper(rawCluster, region);
-        }
-      } else {
-        const out = [];
-        for (const region of enabledRegions) {
-          const client = (await ctx.getAwsClient(region)) as AWS;
-          const rawClusters = (await this.getCacheClusters(client.elasticacheClient)) ?? [];
-          for (const i of rawClusters) {
-            if (i.CacheClusterStatus === 'deleting') continue;
-            const outCacheCluster = this.cacheClusterMapper(i, region);
-            if (outCacheCluster) out.push(outCacheCluster);
+          if (rawCluster && rawCluster.CacheClusterStatus !== 'deleting') {
+            return this.cacheClusterMapper(rawCluster, region);
           }
         }
+      } else {
+        const out: CacheCluster[] = [];
+        await Promise.all(
+          enabledRegions.map(async region => {
+            const client = (await ctx.getAwsClient(region)) as AWS;
+            const rawClusters = (await this.getCacheClusters(client.elasticacheClient)) ?? [];
+            for (const i of rawClusters) {
+              if (i.CacheClusterStatus === 'deleting') continue;
+              const outCacheCluster = this.cacheClusterMapper(i, region);
+              if (outCacheCluster) out.push(outCacheCluster);
+            }
+          }),
+        );
         return out;
       }
     },
     update: async (clusters: CacheCluster[], ctx: Context) => {
       // if user has modified state, restore it. If not, go with replace path
       const out = [];
       for (const cluster of clusters) {
-        const cloudRecord = ctx?.memo?.cloud?.CacheCluster?.[cluster.clusterId ?? ''];
+        const cloudRecord = ctx?.memo?.cloud?.CacheCluster?.[this.entityId(cluster)];
```

Comment:
oh I didn't know this was a thing. this extracts the entity?

Reply:
Yep, if you use `@cloudId`, under the hood it is implementing the `entityId` method for you.